### PR TITLE
Timed output

### PR DESCRIPTION
**Pull request by aditivijayan**
This modification allows users to create data dumps at equal time intervals. Checkpoints are created when plt files are stored.

### CODE REVIEW

**Review by aditivijayan — COMMENTED** [review]
Changes suggested in hydro_system.hpp are superfluous, I believe.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -126,5 +126,3 @@ add_subdirectory(ODEIntegration)
 add_subdirectory(Cooling)
 add_subdirectory(ShockCloud)
 add_subdirectory(PassiveScalar)
-add_subdirectory(FCQuantities)
-add_subdirectory(SphericalCollapse)
```
Comment (BenWibking) [inline src/CMakeLists.txt]: Can you revert this change? These should not be modified

File: src/hydro_system.hpp [hunk 2]
```diff
@@ -939,4 +939,4 @@ void HydroSystem<problem_t>::ComputeFluxes(amrex::MultiFab &x1Flux_mf, amrex::Mu
 	});
 }
 
-#endif // HYDRO_SYSTEM_HPP_
+#endif // HYDRO_SYSTEM_HPP_
```
Comment (BenWibking) [inline src/hydro_system.hpp]: Can you revert this change? There's no reason to modify this file.

File: src/simulation.hpp [hunk 3]
```diff
@@ -703,6 +708,14 @@ template <typename problem_t> void AMRSimulation<problem_t>::evolve()
 			WritePlotFile();
 		}
 
+		// Writing Plot files at time intervals
+		double next_plot_file_time = next_plot_file_num * plotTimeInterval_;
+		if (plotTimeInterval_ > 0 && ((next_plot_file_time > cur_time) && (next_plot_file_time < (cur_time + dt_[0])))) {
```
Comment (BenWibking) [inline src/simulation.hpp]: I think there might be a floating-point roundoff issue here. Is the last condition necessary?

File: src/simulation.hpp [hunk 4]
```diff
@@ -703,6 +708,14 @@ template <typename problem_t> void AMRSimulation<problem_t>::evolve()
...
+			next_plot_file_num += 1;
+			WritePlotFile();
+			WriteCheckpointFile();
```
Comment (BenWibking) [inline src/simulation.hpp:724]: The checkpoint files can be very large. I think we should not automatically output the checkpoint file at the same time. Perhaps make that a separate option?

File: src/simulation.hpp [hunk 5]
```diff
@@ -703,6 +708,14 @@ template <typename problem_t> void AMRSimulation<problem_t>::evolve()
 			WritePlotFile();
 		}
 
+		// Writing Plot files at time intervals
+		double next_plot_file_time = next_plot_file_num * plotTimeInterval_;
+		if (plotTimeInterval_ > 0 && ((next_plot_file_time > cur_time) && (next_plot_file_time < (cur_time + dt_[0])))) {
+			next_plot_file_num += 1;
```
Comment (BenWibking) [inline src/simulation.hpp]: I think this might be clearer if you change it to `next_plot_file_time += plotTimeInterval_`. Then `next_plot_file_num` is not needed.

**Review by BenWibking — CHANGES_REQUESTED** [review]
There are some minor comments. Also, can you revert the changes to the amrex submodule, hydro_system.hpp and src/CMakeLists.txt? It looks like those were added by mistake.

File: src/simulation.hpp (hunk 3 above)
Comment (aditivijayan) [reply src/simulation.hpp]: Do you mean- `(next_plot_file_time < (cur_time + dt_[0])`? Yes, we need to make sure that `next_plot_file_time` is within dt of `cut_time`. Because if `next_plot_file_time` is say cur_time + 3dt, then it would write 3 files instead of 1.

File: src/simulation.hpp (hunk 5 above)
Comment (aditivijayan) [reply src/simulation.hpp]: I can do this.

File: src/simulation.hpp (hunk 3 above)
Comment (BenWibking) [reply src/simulation.hpp]: Could you change the latter two conditions to just `next_plot_file_time <= cur_time`?

File: src/simulation.hpp (hunk 4 above)
Comment (aditivijayan) [reply src/simulation.hpp:724]: Sure thing.

File: src/hydro_system.hpp (hunk 2 above)
Comment (BenWibking) [reply src/hydro_system.hpp]: You can checkout the last commit that modified this file with
```
git checkout `git --no-pager log -1 --pretty=tformat:"%h" -- src/hydro_system.hpp` -- src/hydro_system.hpp
```

Comment (BenWibking) [reply src/hydro_system.hpp]: Ah, oops, that won't work. This should checkout the previous version of the file:
```
git checkout `git --no-pager log -1 --skip 1 --pretty=tformat:"%h" -- src/hydro_system.hpp` -- src/hydro_system.hpp
```

Comment (BenWibking) [reply src/hydro_system.hpp]: Hm, ok, the previous commit doesn't seem to be the same as that on the development branch. It might be better to just checkout the version of this file from the development branch, e.g.:
```
git checkout development -- src/hydro_system.hpp
```

**Comment by BenWibking** [comment]
Something is still not right with hydro_system.hpp. Do you have the latest version pulled locally? You might have to checkout development with `git checkout development` and then `git pull` to make sure you have the latest version of the development branch.

**Comment by aditivijayan** [comment]
> Something is still not right with hydro_system.hpp. Do you have the latest version pulled locally? You might have to checkout development with `git checkout development` and then `git pull` to make sure you have the latest version of the development branch.

Locally, it shows that everything is up to date with the development branch. But I noticed that the declaration of AddinternalpDV was different. So, I've just copy-pasted the right one from the development branch. Hopefully, it should work now.

**Comment by BenWibking** [comment]
> Locally, it shows that everything is up to date with the development branch. But I noticed that the declaration of AddinternalpDV was different. So, I've just copy-pasted the right one from the development branch. Hopefully, it should work now.

Hmm, it's not showing that when I look at the changes here: https://github.com/quokka-astro/quokka/pull/227/files.

**Comment by aditivijayan** [comment]
I am really perplexed by this because I haven't touched that part of the code at all. I've just copied the entire function from the development branch now.

**Comment by BenWibking** [comment]
Ok, that was strange. It looks like it's fixed now.

Can you set the amrex submodule to the same commit used on the development branch?

**Comment by aditivijayan** [comment]
Ohhh.. I don't know how to do that.

**Comment by BenWibking** [comment]
> Ohhh.. I don't know how to do that.

You can check the current commit of the amrex submodule by looking at the development branch for Quokka here: https://github.com/quokka-astro/quokka/tree/development/extern.

Then, on your local machine, go into the submodule directory:
`cd extern/amrex`

Checkout this commit:
`git checkout ae58672`

Go back to the main git repository:
`cd ../..`

`git status` should now show that the submodule has been updated. You can then commit the change to the submodule with `git add extern/quokka && git commit -m 'revert amrex submodule'`.

**Comment by aditivijayan** [comment]
Thanks, Ben!

**Review by BenWibking — APPROVED** [review]

**Comment by BenWibking** [comment]
I'm running the GPU tests now. If those pass, it can be merged.

**Comment by aditivijayan** [comment]
Great! Thanks, Ben!
